DemoApp/Sources/Components/Router.swift

```diff
@@ -76,9 +77,13 @@ final class Router: ObservableObject {
         if AppEnvironment.configuration == .test, AppEnvironment.contains(.mockJWT) {
             return
         } else if let userCredentials = AppState.shared.unsecureRepository.loadCurrentUser() {
-            appState.currentUser = userCredentials.userInfo
-            appState.userState = .loggedIn
-            handleLoggedInUserCredentials(userCredentials, deeplinkInfo: .empty)
+            if userCredentials.userInfo.id.contains("@getstream") {
+                GIDSignIn.sharedInstance.restorePreviousSignIn { [weak self] _,_ in
+                    self?.setupUser(with: userCredentials)
```

Comment:
Is the an error being returned here that should be handling?

Reply:
No, if a google sign in fails to happen, we just setup the user as normal. For example, your id can contain @getstream, but it's not done via Google sign in.